### PR TITLE
[terraform-resources] early exit

### PR DESCRIPTION
**Pull request by maorfr**
the desired state is represented by:
- aws accounts
- namespaces with external resources
- resources referenced from defaults path (or parameter_group for rds).

with no yaml loading, an early exit takes.................. 6 seconds!

### CODE REVIEW

File: reconcile/terraform_resources.py [hunk 1]
```diff
@@ -716,3 +721,25 @@ def run(
...
+
+def early_exit_desired_state(*args, **kwargs) -> dict[str, Any]:
+    gqlapi = gql.get_api()
+    namespaces = get_tf_namespaces()
+    resources = []
+    for ns_info in namespaces:
+        for spec in get_external_resource_specs(
+            ns_info, provision_provider=PROVIDER_AWS
+        ):
+            defaults = spec.resource.get("defaults")
```
Comment (geoberle) [inline reconcile/terraform_resources.py:734]: SQS and dynamodb have their defaults nested

File: reconcile/terraform_resources.py [hunk 2]
```diff
@@ -716,3 +721,25 @@ def run(
...
+            if defaults:
+                resources.append(gqlapi.get_resource(defaults))
```
Comment (geoberle) [inline reconcile/terraform_resources.py:736]: Nit: This is again my opportunity to shamelessly plug resource ref loading 😊

We should consider it for a followup PR

File: reconcile/terraform_resources.py (hunk 1 above)
Comment (maorfr) [reply reconcile/terraform_resources.py:734]: good catch! https://github.com/app-sre/qontract-reconcile/pull/2714/commits/3a50e9c0509c30753d27eab8100e2c3e89a4494d

**Review by geoberle — APPROVED** [review]
I'm not entirely happy with the "hardcoded" default fields but this approach works correct and gets the feature out there way faster than refactoring for resource ref loading. Therefore Kudos for this nice adoption of early exit. LGTM ❤️

**Comment by maorfr** [comment]
https://issues.redhat.com/browse/APPSRE-6098

**Pull request merged**